src/overlays/actors/ovl_En_Takaraya/z_en_takaraya.c

```diff
@@ -28,61 +47,360 @@ ActorInit En_Takaraya_InitVars = {
+        play->transitionTrigger = TRANS_TRIGGER_START;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF050.s")
+void func_80ADF6DC(EnTakaraya* this) {
+    Animation_PlayLoop(&this->skelAnime, &object_bg_Anim_001384);
+    this->eyeTexIndex = 0;
+    this->actor.flags |= ACTOR_FLAG_10000;
+    this->actionFunc = &func_80ADF730;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF2D4.s")
+void func_80ADF730(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
+        this->actor.flags &= 0xFFFEFFFF;
+        func_80ADF7B8(this);
+    } else {
+        this->unk2B0 = gSaveContext.save.playerForm + this->switchFlag;
+        Actor_OfferTalk(&this->actor, play, 120.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF338.s")
+void func_80ADF7B8(EnTakaraya* this) {
+    this->actionFunc = &func_80ADF7CC;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF4E0.s")
+void func_80ADF7CC(EnTakaraya* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF520.s")
+    if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && (Message_ShouldAdvance(play))) {
+        if (this->actor.textId == 0x77A) {
+            if (CHECK_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED)) {
+                Message_CloseTextbox(play);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_WAIT);
+                CLEAR_WEEKEVENTREG(WEEKEVENTREG_KICKOUT_TIME_PASSED);
+                func_80ADEDF8(this);
+            } else {
+                this->actor.textId = 0x77C;
+                Message_ContinueTextbox(play, this->actor.textId);
+            }
+        } else {
+            this->actor.textId = D_80ADFB50[(void)0, gSaveContext.save.playerForm];
+            Message_ContinueTextbox(play, this->actor.textId);
+            Animation_MorphToPlayOnce(&this->skelAnime, &object_bg_Anim_00AD98, 5.0f);
+            func_80ADF03C(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF608.s")
+void EnTakaraya_Update(Actor* thisx, PlayState* play) {
+    EnTakaraya* this = THIS;
+    s32 pad;
+    Vec3s torsoRot;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF654.s")
+    this->actionFunc(this, play);
+    Actor_TrackPlayer(play, &this->actor, &this->headRot, &torsoRot, this->actor.focus.pos);
+    func_80ADED8C(this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF6DC.s")
+s32 EnTakaraya_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnTakaraya* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Takaraya/func_80ADF730.s")
+    if (limbIndex == 5) {
```

Comment:
```suggestion
    if (limbIndex == OBJECT_BG_1_LIMB_05) {
```
And also the same limb is used below